servlet/src/jettyTest/java/io/grpc/servlet/JettyTransportTest.java

```diff
@@ -0,0 +1,249 @@
+            // Although specified here, address is ignored because we never call build.
+            .forAddress("localhost", 0)
+            .flowControlWindow(65 * 1024)
+            .negotiationType(NegotiationType.PLAINTEXT);
+    InternalNettyChannelBuilder
+            .setTransportTracerFactory(nettyChannelBuilder, fakeClockTransportTracer);
+    ClientTransportFactory clientFactory =
+            InternalNettyChannelBuilder.buildTransportFactory(nettyChannelBuilder);
+    return clientFactory.newClientTransport(
+            new InetSocketAddress("localhost", port),
+            new ClientTransportFactory.ClientTransportOptions()
+                    .setAuthority(testAuthority(server))
+                    .setEagAttributes(eagAttrs()),
+            transportLogger());
+  }
+
+  @Override
+  protected String testAuthority(InternalServer server) {
+    return "localhost:" + port;
+  }
+
+  @Override
+  protected void advanceClock(long offset, TimeUnit unit) {
+    fakeClock.forwardNanos(unit.toNanos(offset));
+  }
+
+  @Override
+  protected long fakeCurrentTimeNanos() {
+    return fakeClock.getTicker().read();
+  }
+
+  @Override
+  @Ignore("Skip the test, server lifecycle is managed by the container")
+  @Test
+  public void serverAlreadyListening() {
+  }
+
+  @Override
+  @Ignore("Skip the test, server lifecycle is managed by the container")
+  @Test
+  public void openStreamPreventsTermination() {
+  }
+
+  @Override
+  @Ignore("Skip the test, server lifecycle is managed by the container")
+  @Test
+  public void shutdownNowKillsServerStream() {
+  }
+
+  @Override
+  @Ignore("Skip the test, server lifecycle is managed by the container")
+  @Test
+  public void serverNotListening() {
+  }
+
+  // FIXME
+  @Override
+  @Ignore("Servlet flow control not implemented yet")
+  @Test
+  public void flowControlPushBack() {
+  }
+
+  // FIXME
+  @Override
+  @Ignore("Jetty is broken on client RST_STREAM")
+  @Test
+  public void shutdownNowKillsClientStream() {
+  }
+
+  @Override
+  @Ignore("Server side sockets are managed by the servlet container")
+  @Test
+  public void socketStats() {
+  }
+
+  @Override
+  @Ignore("serverTransportListener will not terminate")
+  @Test
+  public void clientStartAndStopOnceConnected() {
+  }
+
+  @Override
+  @Ignore("clientStreamTracer1.getInboundTrailers() is not null; listeners.poll() doesn't apply")
+  @Test
+  public void serverCancel() {
+  }
+
+  @Override
+  @Ignore("THis doesn't apply: Ensure that for a closed ServerStream, interactions are noops")
+  @Test
+  public void interactionsAfterServerStreamCloseAreNoops() {
+  }
+
+  @Override
+  @Ignore("listeners.poll() doesn't apply")
+  @Test
+  public void interactionsAfterClientStreamCancelAreNoops() {
+  }
+
+  @Override
+  @Ignore("assertNull(serverStatus.getCause()) isn't true")
+  @Test
+  public void clientCancel() {
+  }
+
+  @Override
+  @Ignore("regression since bumping grpc v1.46 to v1.47")
```

Comment:
I filed #9308 to track the issue (which can be referenced here in the annotation). Seems there's no update to the issue. One can bump to the latest grpc version to see if the regression is still there. Otherwise need to do a bisect from v1.45 (the annotation mentioned v1.46 but actually it was bumping from v1.45) to v1.47 to find which change has caused the issue.

Reply:
I've merged up to latest and the test still fails, incremented the version.